tests/tests.py

```diff
@@ -41,33 +43,87 @@ def osm(request):
 
 
 @pytest.fixture
-def writer(osm):
+def dummy_transit_data(transit_data):
+    return \
+        gtfs_dummy.create_dummy_data(transit_data.routes,
+            transit_data.stops,
+            transit_data.route_stops)
+
+
+@pytest.fixture
+def gtfs_writer(transit_data):
     w = GTFSWriter()
-    w.add_agencies(osm.agencies.values())
-    w.add_stops(osm.stops.values())
-    w.add_routes(osm.routes.values())
+
+    w.add_agencies(transit_data.agencies.values())
+    w.add_stops(transit_data.stops.values())
+    w.add_routes(transit_data.routes.values())
 
     return w
 
 
 @pytest.fixture
-def dummy(osm):
-    return \
-        gtfs_dummy.create_dummy_data(osm.routes,
-            osm.stops,
-            osm.route_stops)
+def dummy_gtfs_writer(transit_data, dummy_transit_data):
+    w = GTFSWriter()
 
+    # Patch agencies to pass transitfeed check
+    for agency in transit_data.agencies.values():
+        if 'agency_url' not in agency or not agency['agency_url']:
```

Comment:
@mehdisadeghi shouldnt this be part of the dummy generator itself instead of being a workaround to pass the tests?

Reply:
@carocad That's true. I couldn't think of a better solution for now. The problem is, the dummy generator doesn't touch agencies in order to produce its output. Perhaps I have to pass the agencies and let them be patched there. What do you think?